File: modules/wem-api/src/main/java/com/enonic/wem/api/module/ModuleUpdatedEvent.java
package com.enonic.wem.api.module;

import com.google.common.base.Objects;

import com.enonic.wem.api.event.Event;

public final class ModuleUpdatedEvent
    implements Event
{
    private final ModuleState state;

    private final ModuleKey moduleKey;

    public ModuleUpdatedEvent( final ModuleKey key, final ModuleState state )
    {
        this.state = state;
        this.moduleKey = key;
    }

    public ModuleState getState()
    {
        return state;
    }

    public ModuleKey getModuleKey()
    {
        return moduleKey;
    }

    @Override
    public String toString()
    {
        return Objects.toStringHelper( this ).
            add( "state", this.state ).
            add( "moduleKey", this.moduleKey ).
            omitNullValues().
            toString();
    }
}

File: modules/wem-admin-ui/src/main/resources/web/admin/common/js/module/ModuleUpdatedEvent.ts
module api.module {

    export interface ModuleUpdatedEventJson {
        state: string;
        moduleKey: string;
    }

    export class ModuleUpdatedEvent extends api.event.Event {

        private moduleKey: api.module.ModuleKey;

        private state: string;

        constructor(moduleKey: api.module.ModuleKey, state: string) {
            super();
            this.moduleKey = moduleKey;
            this.state = state;
        }

        public getModuleKey(): api.module.ModuleKey {
            return this.moduleKey;
        }

        public getState(): string {
            return this.state;
        }

        static on(handler: (event: ModuleUpdatedEvent) => void) {
            api.event.Event.bind(api.util.getFullName(this), handler);
        }

        static un(handler?: (event: ModuleUpdatedEvent) => void) {
            api.event.Event.unbind(api.util.getFullName(this), handler);
        }

        static fromJson(json: ModuleUpdatedEventJson): ModuleUpdatedEvent {
            var moduleKey = api.module.ModuleKey.fromString(json.moduleKey);
            var state = json.state;
            return new ModuleUpdatedEvent(moduleKey, state);
        }
    }

}